example/concurrency-03.rst

```diff
@@ -0,0 +1,19 @@
+let count = 0;
+
+fn increment(times: int) {
+  let i = 0;
+  while i < times {
```

Comment:
Syntax wise, for loops I think I will just use the `loop` keyword with optional condition i.e `loop i < times { .. }` to stop after just like how Go has `for` and `for <cond>`.